### PR TITLE
started code review

### PR DESCRIPTION
**Pull request by Snover98**

### CODE REVIEW

File: xml_to_json.py [hunk 1]
```diff
@@ -1,3 +1,96 @@
...
+        data = xml_file.read()
+
+    return xmltodict.parse(data)
+
+
+def convert_xml_file_to_json(xml_file_path: str, delete_xmls: bool):
+    values_dict = xml_path_to_dict(xml_file_path)
+    json_string = json.dumps(values_dict, sort_keys=True, indent=4)
+
+    json_file_path = xml_file_path.split('.')[0] + '.json'
```
Comment (cabir1) [inline xml_to_json.py]: What if the file name includes a dot

File: xml_to_json.py [hunk 2]
```diff
@@ -1,3 +1,96 @@
...
+    if os.path.isdir(path):
+        to_convert = list(all_xmls(path, recursion_depth))
+
+        if len(to_convert) == 0:
+            raise Exception("ERROR: No XML files found!")
+
+        pool = Pool(num_workers)
+        pbar = tqdm(desc="XML to JSON Conversion Progress", total=len(to_convert))
+        for xml_file in to_convert:
+            pool.apply_async(convert_xml_file_to_json, args=(xml_file, delete_xmls), callback=lambda _: pbar.update())
```
Comment (cabir1) [inline xml_to_json.py:81]: I’m not familiar with pool but from a quick look you turn it on the same file?!

**Review by cabir1 — COMMENTED** [review]
Overall great script

File: xml_to_json.py (hunk 1 above)
Comment (Snover98) [reply xml_to_json.py]: valid file names cannot contain a dot except for the extension.
if a directory contains a dot it's hidden and we should not touch it anyways

File: xml_to_json.py (hunk 2 above)
Comment (Snover98) [reply xml_to_json.py:81]: nope, for each file I'm adding a task to the pool for converting it

File: xml_to_json.py (hunk 1 above)
Comment (cabir1) [reply xml_to_json.py]: You can contain dot in a file/folder name

Comment (Snover98) [reply xml_to_json.py]: fixed

**Pull request merged**